File: 2025/04 April/24 Unique Number III/main.py
# Link: https://www.geeksforgeeks.org/problems/find-element-occuring-once-when-all-other-are-present-thrice/1


class Solution:
    def getSingle(self, arr):
        freq = {}
        for num in arr:
            if num in freq:
                freq[num] += 1
            else:
                freq[num] = 1
        for num in freq:
            if freq[num] == 1:
                return num
        return -1


if __name__ == "__main__":
    obj = Solution()
    arr = [1, 10, 1, 1]
    print(obj.getSingle(arr))

    arr = [3, 2, 1, 34, 34, 1, 2, 34, 2, 1]
    print(obj.getSingle(arr))

File: 2025/04 April/24 Unique Number III/main.cpp
// Link: https://www.geeksforgeeks.org/problems/find-element-occuring-once-when-all-other-are-present-thrice/1

#include <iostream>
#include <vector>
#include <map>
using namespace std;

class Solution
{
public:
    int getSingle(vector<int> &arr)
    {
        map<int, int> freq;
        for (auto it : arr)
            freq[it]++;

        vector<int> ans;
        for (auto it : freq)
        {
            if (it.second == 1)
                return it.first;
        }
        return -1;
    }
};

int main()
{
    Solution obj;
    vector<int> arr = {1, 10, 1, 1};
    cout << obj.getSingle(arr) << endl;

    arr = {3, 2, 1, 34, 34, 1, 2, 34, 2, 1};
    cout << obj.getSingle(arr) << endl;
    return 0;
}